libr/asm/asm.c

```diff
@@ -117,40 +120,45 @@ static inline int r_asm_pseudo_byte(RAsmOp *op, char *input) {
 	r_str_replace_char (input, ',', ' ');
 	len = r_str_word_count (input);
 	r_str_word_set0 (input);
+	ut8 *buf = malloc (len);
+	if (!buf) {
+		return 0;
+	}
 	for (i = 0; i < len; i++) {
 		const char *word = r_str_word_get0 (input, i);
 		int num = (int)r_num_math (NULL, word);
-		op->buf[i] = num;
+		buf[i] = num;
 	}
-	r_hex_bin2str (op->buf, len, op->buf_hex);
+	r_asm_op_set_buf (op, buf, len);
+	free (buf);
 	return len;
 }
 
 static inline int r_asm_pseudo_fill(RAsmOp *op, char *input) {
-	int i, repeat=0, size=0, value=0;
+	int i, repeat = 0, size=0, value=0;
 	sscanf (input, "%d,%d,%d", &repeat, &size, &value); // use r_num?
 	size *= repeat;
 	if (size > 0) {
+		ut8 *buf = malloc (size);
 		for (i = 0; i < size; i++) {
-			op->buf[i] = value;
+			buf[i] = value;
```

Comment:
Here value can be of the size more than one byte, which is `buf`, so values can overlap. So this function will not work properly on values > 0xff.

Reply:
open an issue for this, its "unrelated" to the pr